db-scheduler/src/test/resources/mariadb_tables.sql

```diff
@@ -11,7 +11,9 @@ create table test.scheduled_tasks (
   consecutive_failures INT,
   last_heartbeat timestamp(6) null,
   version BIGINT not null,
+  priority INT,
   PRIMARY KEY (task_name, task_instance),
   INDEX execution_time_idx (execution_time),
-  INDEX last_heartbeat_idx (last_heartbeat)
+  INDEX last_heartbeat_idx (last_heartbeat),
+  INDEX priority_execution_time_idx (priority desc, execution_time asc)
```

Comment:
I am starting to believe (execution, priority) might give better results. We have to see what the testing shows

Reply:
I'm getting down to testing, I'll be back soon with the data :)